src/OnboardingSPA/components/Sidebar/components/Customize/DesignColorsPanel/index.js

```diff
@@ -168,20 +223,29 @@ const DesignColorsPanel = forwardRef(
 				return;
 			}
 
-			colorPalettes[ selectedPalette ].primary = selectedColor.primary;
-			if ( colorPalettes[ selectedPalette ].secondary ) {
-				colorPalettes[ selectedPalette ].secondary =
+			if ( selectedPalette === 'custom' ) {
+				currentData.sitegen.homepages.data[ slug ].color.customColors =
+					selectedColor;
+			}
```

Comment:
Based on previous comments we should be able to eliminate the extra key `customColors`